bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -1186,16 +1196,31 @@ void destroy () {
 
 private void destroyHandle () {
 	for (Long handle : zoomLevelToHandle.values()) {
-		if (type == SWT.ICON) {
-			OS.DestroyIcon (handle);
-		} else {
-			OS.DeleteObject (handle);
-		}
+		destroyHandle(handle);
 	}
 	zoomLevelToHandle.clear();
 	handle = 0;
 }
 
+@Override
+void destroyHandlesExcept(Set<Integer> zoomLevels) {
+	zoomLevelToHandle.entrySet().removeIf(entry -> {
+		if (!zoomLevels.contains(entry.getKey()) && entry.getKey() != DPIUtil.getZoomForAutoscaleProperty(initialNativeZoom)) {
```

Comment:
Extracting reused `entry.getKey()` into a self-explaining variable would improve readability.

Reply:
Done

Reply:
Pleas also adapt the other places with the same code consistently